### ISSUE TITLE
unable get type of message

### ISSUE THREAD

**Issue by wbm-sbasler** [commercetools/commercetools-sdk-php-v2#89]
**Describe the bug**
When I read the messages from the messages endpoint, I get a list of messages but cant differentiate them, because `getType` always returns `null` although the `MessageModel` has the correct value in `self::FIELD_TYPE`.

**To Reproduce**
```
$messages = $this->apiRoot
    ->messages()
    ->get()
    ->execute();

/** @var MessageModel $message */
foreach ($messages->getResults() as $message) {
    dump($message->getType());
}
```

**Expected behavior**
I expect that `getType` returns the type of the message.

**Screenshots/Code snippet**
see steps to reproduce

**Stack information (please complete the following information):**
 - PHP: [8.1]
 - SDK: [v6.0.1]

**Additional context**
none

**Comment by wbm-sbasler** [comment]
I found out that type is initialized with empty string, so the condition `is_null` in function `getType` will never be true.

**Possbile fixes:**

1. Initialize type with null:
https://github.com/commercetools/commercetools-sdk-php-v2/blob/master/lib/commercetools-api/src/Models/Message/MessageModel.php#L80
`protected $type = null;`

2. change chondition to empty:
https://github.com/commercetools/commercetools-sdk-php-v2/blob/master/lib/commercetools-api/src/Models/Message/MessageModel.php#L405
`if (empty($this->type)) {`

**Comment by jenschude** [comment]
Thanks for the report. I'm already working on a fix for this particular issue. It has to be fixed in the code generator as it touches the way the model classes are built.

**Comment by jenschude** [comment]
I just released 6.0.2 which fixes the issue for the MessageModel class.

The next minor release will fix it also for other discriminating model classes. Will also take a look to instantiate the correct child class inside collections when dealing with a discriminator model
